### PR TITLE
executor: fix show database like case sensitive issue#34766

### LINKED ISSUE #34766: SHOW DATABASES LIKE should be case insensitive
### 1. Minimal reproduce step 
```sql
DROP DATABASE IF EXISTS `TEST_$1`;
DROP DATABASE IF EXISTS `test_$2`;

CREATE DATABASE `TEST_$1`;
CREATE DATABASE `test_$2`;
SHOW DATABASES LIKE "TEST%";
SHOW DATABASES LIKE "test%";

DROP DATABASE `TEST_$1`;
DROP DATABASE `test_$2`;
```

### 2. What did you expect to see? 
```sql
mysql> SHOW DATABASES LIKE "TEST%";
+------------------+
| Database (TEST%) |
+------------------+
| TEST_$1          |
| test_$2          |
+------------------+
2 rows in set (0.00 sec)

mysql> SHOW DATABASES LIKE "test%";
+------------------+
| Database (test%) |
+------------------+
| TEST_$1          |
| test_$2          |
+------------------+
2 rows in set (0.00 sec)
```
### 3. What did you see instead 
```sql
SHOW DATABASES LIKE "TEST%";
+----------+
| Database |
+----------+
| TEST_$1  |
+----------+
1 row in set (0.00 sec)

mysql> SHOW DATABASES LIKE "test%";
+----------+
| Database |
+----------+
| test_$2  |
+----------+
1 row in set (0.00 sec)
```

### 4. What is your TiDB version? 
```
tidb_version(): Release Version: v6.1.0-alpha-460-g2d44ac927
Edition: Community
Git Commit Hash: 2d44ac9274b4694ed237fc14e5d6500398e93164
Git Branch: master
UTC Build Time: 2022-05-19 04:40:29
GoVersion: go1.18.1
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

### PR DESCRIPTION
**Pull request by e11jah**
Signed-off-by: elijah <f1renze.142857@gmail.com>

### What problem does this PR solve?

Issue Number: close #34766

Problem Summary: show database like should not be case sensitive

### What is changed and how it works?

add a filter `plannercore.ShowDatabaseExtractor` in the `ShowExec`. It will bring the like expression to the place where to get meta and filter them for getting the right result.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
```shell
MySQL root@(none):(none)> CREATE DATABASE `TEST_$1`;
Query OK, 1 row affected
Time: 0.002s
MySQL root@(none):(none)> CREATE DATABASE `test_$2`;
Query OK, 1 row affected
Time: 0.001s
MySQL root@(none):(none)> SHOW DATABASES LIKE "TEST_%";
+------------------+
| Database (TEST%) |
+------------------+
| TEST_$1          |
| test_$2          |
+------------------+

2 rows in set
Time: 0.014s
MySQL root@(none):(none)> SHOW DATABASES LIKE "test_%";
+------------------+
| Database (test%) |
+------------------+
| TEST_$1          |
| test_$2          |
+------------------+
2 rows in set
Time: 0.008s
```
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
executor: fix show database like case sensitive issue
```

### CODE REVIEW

**Review by hawkingrei — APPROVED** [review]

File: executor/show.go [hunk 1]
```diff
@@ -402,11 +402,33 @@ func (e *ShowExec) fetchShowDatabases() error {
...
+		}
+		FieldFilterEnable = extractor.Field != ""
+		fieldFilter = extractor.Field
+	}
 	// let information_schema be the first database
 	moveInfoSchemaToFront(dbs)
 	for _, d := range dbs {
 		if checker != nil && !checker.DBIsVisible(e.ctx.GetSessionVars().ActiveRoles, d) {
 			continue
+		} else if FieldFilterEnable && lowerCaseNames[d] != fieldFilter {
```
Comment (xiongjiwei) [inline executor/show.go:428]: why not use `strings.ToLower()` to replace the map

Comment (e11jah) [reply executor/show.go]: fixed

**Review by xiongjiwei — APPROVED** [review]

**Pull request merged**